Tactics/CSE.lean

```diff
@@ -228,58 +272,81 @@ partial def CSEM.planCSE (e : Expr): CSEM GeneralizeArg := do
 Plan to perform a CSE for this expression, by building a 'GeneralizeArg'.
 -/
 def CSEM.generalize (arg : GeneralizeArg) : CSEM Bool := do
+  consumeFuelEliminate
+  unless (← hasFuelEliminate?) do
+    trace[Tactics.cse.summary] "⏸️ CSE ran out of fuel while performing subexpression elimination. increase `fuelEliminate`."
+    return false
```

Comment:
It seems we pretty much only call `consumeFuel` and `hasFuel` together. We could slightly simplify these callsites by having a single function that consumes a fuel, and returns whether there is any left at the same time, at the cost of this function being slightly less self-explanatory than the current split in two functions.

Just a thought, not sure it matters much either way

Reply:
Yeah, I had trouble coming up with a sufficiently clean yet self-explanatory name for the combined method, which is a good hint the functionality might be too magical. Let's leave it as-is then, the duplication is not too bad